app/services/katello/rhsm_fact_parser.rb

```diff
@@ -73,7 +73,13 @@ def operatingsystem
           os_attributes[:name] = "CentOS"
         end
 
-        ::Operatingsystem.find_or_create_by(os_attributes)
```

Comment:
```
begin
  ::Operatingsystem.find_or_create_by(os_attributes)
rescue ActiveRecord::RecordNotUnique
  retry
end
```
would this work?

Reply:
Maybe. I think that may be a less-explicit way of writing the same thing I have there. But I kind of like explicit.. :thinking: